apps/sync/js/adapters/bookmarks.js

```diff
@@ -0,0 +1,347 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+  Bookmarks Data Adapter
+
+    The purpose of the Bookmarks adapter is to retrieve bookmark records from
+  Firefox Sync and merge them in a local bookmarks DataStore
+  ('firefox-sync-bookmarks') which will be hosting Firefox Sync records only.
```

Comment:
I know I proposed this naming, but thinking about twice it's probably better to keep the same naming as 'bookmarks_store', so this could be 'sync_bookmarks_store'.